huaweicloud/services/acceptance/iam/resource_huaweicloud_identity_group_role_assignment_test.go

```
@@ -0,0 +1,295 @@
+package iam
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk/openstack/identity/v3.0/eps_permissions"
+	"github.com/chnsz/golangsdk/openstack/identity/v3/roles"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/iam"
+)
+
+func getIdentityGroupRoleAssignmentResourceFunc(c *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	identityClient, err := c.IdentityV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating IAM v3 client: %s", err)
+	}
+
+	iamClient, err := c.IAMV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating IAM v3.0 client: %s", err)
+	}
+
+	groupID := state.Primary.Attributes["group_id"]
+	roleID := state.Primary.Attributes["role_id"]
+	domainID := state.Primary.Attributes["domain_id"]
+	projectID := state.Primary.Attributes["project_id"]
+	enterpriseProjectID := state.Primary.Attributes["enterprise_project_id"]
+
+	if domainID != "" {
+		return iam.GetGroupRoleAssignmentWithDomainID(identityClient, groupID, roleID, domainID)
+	}
+
+	if projectID != "" {
+		return iam.GetGroupRoleAssignmentWithProjectID(identityClient, groupID, roleID, projectID)
+	}
+
+	if enterpriseProjectID != "" {
+		return iam.GetGroupRoleAssignmentWithEpsID(iamClient, groupID, roleID, enterpriseProjectID)
+	}
+
+	return nil, nil
```

shall we return an error here?

---

fixed